src/unix.rs

```diff
@@ -569,4 +582,48 @@ mod tests {
             assert_eq!(sched_attr.sched_period, 100 * 10_u64.pow(6));
         }
     }
+
+    #[test]
+    #[cfg(target_os = "linux")]
+    fn set_deadline_flags() {
+        // allow the identity operation for clarity
+        #![allow(clippy::identity_op)]
+
+        assert!(set_thread_priority_and_policy(
+            0, // current thread
+            ThreadPriority::Deadline(
+                1 * 10_u64.pow(6),
+                10 * 10_u64.pow(6),
+                100 * 10_u64.pow(6),
+                Some(DeadlineFlags::DeadlineOverrun)
+            ),
+            ThreadSchedulePolicy::Realtime(RealtimeThreadSchedulePolicy::Deadline)
+        )
+        .is_ok());
+
+        // now we check the return values
+        unsafe {
+            let mut sched_attr = SchedAttr::default();
+            let ret = libc::syscall(
+                libc::SYS_sched_getattr,
```

Comment:
We have already used this syscall twice. Perhaps, there is also use of it for others in the library? Can we extract it into a `unix::get_scheduling_attributes` or anything like that, or whatever you think is worth doing?